src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -793,9 +793,21 @@ export default class Fms {
      * @return {boolean}
      */
     replaceDepartureProcedure(routeString) {
-        const routeStringElements = routeString.split(PROCEDURE_OR_AIRWAY_SEGMENT_DIVIDER);
-
-        if (routeStringElements.length !== 3) {
+        var routeStringElements = routeString.split(PROCEDURE_OR_AIRWAY_SEGMENT_DIVIDER);
+
+        // if routeString contains an entry element, use that as the departure runway
+        // otherwise, keep the departure runway already defined
+        // complain if there is no departure runway defined
+        if (routeStringElements.length === 2 ) {
```

Comment:
If like for you to look at this whole block again and try to simplify it.  One thing you'll notice in the codebase, is we try very hard to avoid `else` and especially `else if`.  When logic is structured cleanly, there is rarely a need for either.  If you have questions on how or what, please feel free to reach out.

Also be sure you have eslint installed in your editor.  there are some code style errors here that do not conform.

Reply:
So a question I have is what I can expect from the user. If I can assume either a two or three element entry, one of those branches can go away, but I don't know what's a more desirable command syntax:

`sid SID.EXIT` or `sid ENTRY.SID.EXIT`.

I prefer the former myself, but it leads to a second question - at some point the entry will have to be found and then preppended to the route. At which level is it best to do this - in `AircraftCommander`, `Pilot` or `Fms` ?